app/main.py

```diff
@@ -2,0 +1,37 @@
+class Animal:
+    alive = []
+
+    def __init__(self,
+                 name: str,
+                 health: int = 100,
+                 hidden: bool = False
+                 ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def death(self) -> None:
+        if self in Animal.alive:
+            Animal.alive.remove(self)
```

Comment:
Code Efficiency issue: Use the class name to get access to the class attributes. Replace `self.alive.remove(self)` with `Animal.alive.remove(self)`.